### ISSUE TITLE
The issue of checking the validity period format during certificate verification

### ISSUE THREAD

**Issue by dulanshuangqiao** [pyca/cryptography#12109]
When using Cryptography for certificate verification, Cryptography chooses to check the format of the validity and throws an error according to RFC5280.
![image](https://github.com/user-attachments/assets/50e1cce6-f3d2-4834-b9a4-b156d76940be)
RFC5280 stipulates: CAs conforming to this profile MUST always encode certificate validity dates through the year 2049 as UTCTime; certificate validity dates in 2050 or later MUST be encoded as GeneralizedTime.
OpenSSL developers mentioned
RFC5280 also explicitly states if you keep reading "Conforming applications MUST be able to process validity dates that are encoded in either UTCTime or GeneralizedTime."
The restriction stated on encoding is a CA-only restriction and as such if verify did not support both time formats, then verify would not conform to RFC5280.
He emphasized that the restrictions mentioned in RFC5289 are only applicable to CAs,So he didn't check the format.

**Comment by alex** [comment]
@woodruffw thoughts?

**Comment by woodruffw** [comment]
My interpretation of those two sentences is:

1. Certificates with validity dates through 2049 **MUST** have their validities encoded as UTCTime; certificates with validity dates afterwards **MUST** have their validities encoded as GeneralizedTime.
2. X.509 validators must be able to handle both kinds of dates.

I don't believe these statements are in contradiction: my interpretation is that they're meant to combine such that conformant validators **MUST** be able to handle validities beyond 2049, not that conformant validators must accept incorrectly encoded validities.

In other words: the two statements don't contradict (IMO), they only prevent an X.509 validator from being considered "conforming" when it only supports dates that can be expressed in a UTCTime.

**Comment by alex** [comment]
This interpretation makes sense to me. As such, there's no action to be taken here.

Thanks
